### ISSUE TITLE
Implement the `sqlite3_stmt_status` interface?

### ISSUE THREAD

**Issue by fractaledmind** [sparklemotion/sqlite3-ruby#410]
For developers looking to investigate whether or not their database schema is making an appropriate use of indexes, they need information about the actual usage of the database and its queries. SQLite does provide some historical information about "the performance characteristics of the prepared statements" via the `sqlite3_stmt_status` interface:

> Each prepared statement maintains various [SQLITE_STMTSTATUS counters](https://www.sqlite.org/c3ref/c_stmtstatus_counter.html) that measure the number of times it has performed specific operations. These counters can be used to monitor the performance characteristics of the prepared statements. For example, if the number of table steps greatly exceeds the number of table searches or result rows, that would tend to indicate that the prepared statement is using a full table scan rather than an index.
>
> This interface is used to retrieve and reset counter values from a [prepared statement](https://www.sqlite.org/c3ref/stmt.html). The first argument is the prepared statement object to be interrogated. The second argument is an integer code for a specific [SQLITE_STMTSTATUS counter](https://www.sqlite.org/c3ref/c_stmtstatus_counter.html) to be interrogated. The current value of the requested counter is returned. If the resetFlg is true, then the counter is reset to zero after this interface call returns.

from: https://www.sqlite.org/c3ref/stmt_status.html

A quick code search shows that this C function is not present in this code base, meaning that there is no way for Ruby developers to get at this information. This is valuable information. Could we implement a bridge to provide this information to the Ruby interface?

**Comment by flavorjones** [comment]
@fractaledmind Thanks for the feature suggestion.

It would be really helpful if you would spend sometime designing the Ruby API that you wish existed, and would write (failing) tests that demonstrate the Ruby API you want. Demonstrating how this feature would be used in code is very valuable and would save me hours of time researching and designing an API myself. (For context, my OSS backlog feels a bit overwhelming to me at the moment.)

Once an API has been fleshed out, and I understand how you expect it to work, I may be able to quickly draft a C implementation for you to kick the tires on.

**Comment by fractaledmind** [comment]
Absolutely. And I opened the issue just to put a flag to help me remember this thought. I will take full responsibility for exploring a Ruby interface and provide failing tests to back this interface. 

Your OSS work is valuable, but I never want you to feel obligated to anything from me. Let me finish this message by saying explicitly, I am very grateful for your presence in the Ruby ecosystem and I have found you to be a truly top-tier OSS maintainer and contributor. Thank you for everything you do
